auth0/v3/authentication/token_verifier.py

```diff
@@ -340,3 +339,135 @@ def _verify_payload(self, payload, nonce=None, max_age=None, organization=None):
+
+        return payload
+
+    def _verify_payload(self, payload, scopes=None, permissions=None):
+        try:
+            # on Python 2.7, 'str' keys as parsed as 'unicode'
+            # But 'unicode' was removed on Python 3.7
+            # noinspection PyUnresolvedReferences
+            ustr = unicode
+        except NameError:
+            ustr = str
+
+        # Issuer
+        if 'iss' not in payload or not isinstance(payload['iss'], (str, ustr)):
+            raise TokenValidationError('Issuer (iss) claim must be a string present in the access token')
+        if payload['iss'] != self.iss:
+            raise TokenValidationError(
+                'Issuer (iss) claim mismatch in the access token; expected "{}", '
+                'found "{}"'.format(self.iss, payload['iss']))
+
+        # Subject
+        if 'sub' not in payload or not isinstance(payload['sub'], (str, ustr)):
+            raise TokenValidationError('Subject (sub) claim must be a string present in the access token')
+
+        # Audience
+        if 'aud' not in payload or not (isinstance(payload['aud'], (str, ustr)) or isinstance(payload['aud'], list)):
+            raise TokenValidationError(
+                'Audience (aud) claim must be a string or array of strings present in the access token')
+
+        if isinstance(payload['aud'], list) and self.aud not in payload['aud']:
+            payload_audiences = ", ".join(payload['aud'])
+            raise TokenValidationError(
+                'Audience (aud) claim mismatch in the access token; expected "{}" but was '
+                'not one of "{}"'.format(self.aud, payload_audiences))
+        elif isinstance(payload['aud'], (str, ustr)) and payload['aud'] != self.aud:
+            raise TokenValidationError(
+                'Audience (aud) claim mismatch in the access token; expected "{}" '
+                'but found "{}"'.format(self.aud, payload['aud']))
+
+        # --Time validation (epoch)--
+        now = self._clock or time.time()
+        leeway = self.leeway
+
+        # Expires at
+        if 'exp' not in payload or not isinstance(payload['exp'], int):
+            raise TokenValidationError('Expiration Time (exp) claim must be a number present in the access token')
+
+        exp_time = payload['exp'] + leeway
+        if now > exp_time:
+            raise TokenValidationError(
+                'Expiration Time (exp) claim error in the access token; current time ({}) is '
+                'after expiration time ({})'.format(now, exp_time))
+
+        # Issued at
+        if 'iat' not in payload or not isinstance(payload['iat'], int):
+            raise TokenValidationError('Issued At (iat) claim must be a number present in the access token')
+
+        # Scopes
+        if scopes:
+            if 'scope' not in payload or not isinstance(payload['scope'], (str, ustr)):
+                raise TokenValidationError('No scopes claim found in token.')
```

Comment:
Is this claim required per the spec or Auth0's implementation? Seems like instead, we should only be failing if this is not set if we were given an expected value for it in the first place. 

The message should be similar to the rest. Something like `{claim} ({claim short name}) claim must be a {claim type} present in the access token`.